pkg/virt-handler/heartbeat/multipath-socket.go

```diff
@@ -0,0 +1,77 @@
+package heartbeat
+
+import (
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/fsnotify/fsnotify"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/storage/reservation"
+)
+
+const (
+	runDir              = "/proc/1/root/run"
+	socket              = "multipathd.socket"
+	multipathSocketPath = runDir + "/" + socket
+)
+
+type MonitorMultipathSocket struct {
+	w *fsnotify.Watcher
+}
+
+func NewMonitorMultipathSocket() *MonitorMultipathSocket {
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed creating watcher for multipath socket")
+	}
+	w.Add(runDir)
+	return &MonitorMultipathSocket{w: w}
+}
+
+func (m *MonitorMultipathSocket) Run() error {
+	dir := reservation.GetPrHelperHostSocketDir()
+	path := filepath.Join(dir, socket)
+
+	if _, err := os.Stat(dir); errors.Is(err, os.ErrNotExist) {
+		if err := os.Mkdir(dir, 0700); err != nil {
+			log.Log.Reason(err).Errorf("failed creating persistent reservation directory %s", dir)
+		}
+	}
+	if _, err := os.Stat(multipathSocketPath); !errors.Is(err, os.ErrNotExist) {
+		// Remove the hardlink in case it already exists and recreates it
+		os.Remove(path)
+		if out, err := exec.Command("ln", multipathSocketPath, path).CombinedOutput(); err != nil {
```

Comment:
Let's just explain somewhere why hard link was chosen over symlink

Reply:
I did explain why we cannot access `/proc/<pid>/root` in the commit message, but didn't mentioned explicitly the symlinks. I can add a note there.